profiler/src/main/java/com/splunk/opentelemetry/profiler/exporter/PprofProfilingEventExporter.java

```diff
@@ -0,0 +1,253 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.splunk.opentelemetry.profiler.exporter;
+
+import static com.splunk.opentelemetry.profiler.LogExporterBuilder.INSTRUMENTATION_LIBRARY_INFO;
+import static com.splunk.opentelemetry.profiler.ProfilingSemanticAttributes.PROFILING_SOURCE;
+import static com.splunk.opentelemetry.profiler.ProfilingSemanticAttributes.SOURCE_EVENT_NAME;
+import static com.splunk.opentelemetry.profiler.ProfilingSemanticAttributes.SOURCE_EVENT_PERIOD;
+import static com.splunk.opentelemetry.profiler.ProfilingSemanticAttributes.SOURCE_TYPE;
+import static com.splunk.opentelemetry.profiler.pprof.PprofAttributeKeys.DATA_FORMAT;
+import static com.splunk.opentelemetry.profiler.pprof.PprofAttributeKeys.DATA_TYPE;
+
+import com.google.perftools.profiles.ProfileProto.Label;
+import com.google.perftools.profiles.ProfileProto.Sample;
+import com.splunk.opentelemetry.profiler.Configuration.DataFormat;
+import com.splunk.opentelemetry.profiler.context.StackToSpanLinkage;
+import com.splunk.opentelemetry.profiler.events.EventPeriods;
+import com.splunk.opentelemetry.profiler.pprof.Pprof;
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.api.trace.SpanContext;
+import io.opentelemetry.sdk.logs.LogProcessor;
+import io.opentelemetry.sdk.logs.data.LogDataBuilder;
+import io.opentelemetry.sdk.resources.Resource;
+import java.nio.charset.StandardCharsets;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class PprofProfilingEventExporter implements ProfilingEventExporter {
+  private static final Logger logger = LoggerFactory.getLogger(PprofProfilingEventExporter.class);
+
+  private final LogProcessor logProcessor;
+  private final Resource resource;
+  private final DataFormat profilingDataFormat;
+  private final EventPeriods eventPeriods;
+  private Pprof pprof = createPprof();
+
+  private PprofProfilingEventExporter(Builder builder) {
+    this.logProcessor = builder.logProcessor;
+    this.resource = builder.resource;
+    this.profilingDataFormat = builder.profilingDataFormat;
+    this.eventPeriods = builder.eventPeriods;
+  }
+
+  @Override
+  public void export(StackToSpanLinkage stackToSpanLinkage) {
+    StackTrace st = handleStackTrace(stackToSpanLinkage.getRawStack());
+    if (st.stackTraceLines.isEmpty()) {
+      return;
+    }
+
+    Sample.Builder sample = Sample.newBuilder();
+    {
+      Label.Builder label = Label.newBuilder();
+      // XXX should we attempt to extract relevant attributes here?
```

Comment:
It would be nice to split it up into `thread.name`, `thread.id` and `thread.status`, like we have for allocations (well, except for the status), for the sake of consistency.